### DISCUSSION TITLE
Steps to do an instalation on ubuntu

### DISCUSSION THREAD

**Discussion by oscarcarserud** [vlachoudis/bCNC#1669] [Show and tell]
sudo su
apt-get install python3-tk
apt install puton3-pip
pip install --upgrade bCNC

You know, if you do not have python beforehand it is hard to get it working.

**Comment by Harvie** [comment]
> You know, if you do not have python beforehand it is hard to get it working.

Well. This is already explained on wiki: https://github.com/vlachoudis/bCNC/wiki/Installation
Specificaly in these words:

>> For both Python versions, you need tkinter support and working pip. Then, you can use pip install bCNC to install bCNC and all of it's dependencies (except for tkinter, which has to be installed/compiled as an integral part of python).

**Comment by oscarcarserud** [comment]
Yes of course, I understand your point. But then again. IF there are
DEPENDENCIES. I think a decent installer would resolve that. not shit on
it. And I think that "by the way if you did not install python beforehand,
you did not install it and therefore it is not installed," is a kind of
oxymoron.  Better then with a guide how to install a program that WORKS
without googling for answers for 0.5 days for a language you do not use and
do not intend to use and whose specific you are not utterly interested in.

Do you get my point?

Den tis 1 feb. 2022 kl 10:23 skrev Tomas Mudrunka ***@***.***
>:

> You know, if you do not have python beforehand it is hard to get it
> working.
>
> Well. This is already explained on wiki:
> https://github.com/vlachoudis/bCNC/wiki/Installation
> Specificaly in these words:
>
> For both Python versions, you need tkinter support and working pip. Then,
> you can use pip install bCNC to install bCNC and all of it's dependencies
> (except for tkinter, which has to be installed/compiled as an integral part
> of python).
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/vlachoudis/bCNC/discussions/1669#discussioncomment-2087443>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABQB4U52L3M5SMHMKGMPCKLUY6RA7ANCNFSM5NHEHDGQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Detta meddelande är konfidentiellt och får inte återges i någon form, vare
sig i delar eller i dess helhet.

**Comment by Harvie** [comment]
> I think a decent installer would resolve that

Yes. Pip is decent installer which resolves dependencies.

**Comment by jsiddall** [comment]
I also came across a pil dependency so I added the packages I needed to install in Ubuntu to the wiki Linux install procedure.  It will vary a bit by distribution but at least it is called out now.
